### PR TITLE
add database description in dbmanager plugin

### PR DESCRIPTION
**Pull request by npltr62**
This PR provide the oppotinity to display database comment.

### CODE REVIEW

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 1]
```diff
@@ -33,19 +33,25 @@ def __init__(self, db):
...
+            (QApplication.translate("DBManagerPlugin", "Server version: "), info[0]),
         ]
         return HtmlTable(tbl)
 
     def connectionDetails(self):
         tbl = [
             (QApplication.translate("DBManagerPlugin", "Host:"), self.db.connector.host),
-            (QApplication.translate("DBManagerPlugin", "User:"), self.db.connector.user)
-        ]
+            (QApplication.translate("DBManagerPlugin", "User:"), self.db.connector.user)        ]
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/info_model.py:54]:
```suggestion
            (QApplication.translate("DBManagerPlugin", "User:"), self.db.connector.user),
        ]
```

format

**Review by nicogodet — CHANGES_REQUESTED** [review]
General comment: u string are not needed in Python3. See https://github.com/qgis/QGIS/pull/51167

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 2]
```diff
@@ -396,10 +395,18 @@ def fieldTypes(self):
             "date", "time", "timestamp",  # date/time
             "boolean"  # bool
         ]
-
+        
+    def getDatabaseInfo(self):
+        """ db privileges: (can create schemas, can create temp. tables) """
+        sql = u"SELECT pg_catalog.shobj_description( d.oid, 'pg_database') AS description from pg_catalog.pg_database d where  datname = current_database()"
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/postgis/connector.py:401]:
```suggestion
        sql = "SELECT pg_catalog.shobj_description(d.oid, 'pg_database') AS description from pg_catalog.pg_database d where datname = current_database()"
```

format

**Comment by nicogodet** [comment]
I don't what you used to format your code but you added u string everywhere which is not needed in python3. You have to remove them.

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 3]
```diff
@@ -33,11 +33,18 @@ def __init__(self, db):
 
     def __del__(self):
         self.db = None
-
+
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/info_model.py:36]:
```suggestion

```

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 4]
```diff
@@ -33,11 +33,18 @@ def __init__(self, db):
...
+        tbl = [
+            (QApplication.translate("DBManagerPlugin", "Database description: "), info[0])
+        ]
+        return HtmlTable(tbl)
+        
     def generalInfo(self):
         info = self.db.connector.getInfo()
         tbl = [
-            (QApplication.translate("DBManagerPlugin", "Server version: "), info[0])
+            (QApplication.translate("DBManagerPlugin", "Server version: "), info[0]),
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/info_model.py:47]:
```suggestion
            (QApplication.translate("DBManagerPlugin", "Server version: "), info[0])
```

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 5]
```diff
@@ -280,7 +293,7 @@ def triggersDetails(self):
         # add table contents
         for trig in self.table.triggers():
             name = '%(name)s (<a href="action:trigger/%(name)s/%(action)s">%(action)s</a>)' % {"name": trig.name,
-                                                                                               "action": "delete"}
+                                                                                                "action": "delete"}
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/info_model.py:296]:
```suggestion
                                                                                               "action": "delete"}
```

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 6]
```diff
@@ -385,7 +398,7 @@ def spatialInfo(self):
         sr_info = self.table.database().connector.getSpatialRefInfo(srid) if srid != -1 else QApplication.translate(
             "DBManagerPlugin", "Undefined")
         if sr_info:
-            tbl.append((QApplication.translate("DBManagerPlugin", "Spatial ref:"), "%s (%d)" % (sr_info, srid)))
+            tbl.append((QApplication.translate("DBManagerPlugin", "Spatial ref:"), u"%s (%d)" % (sr_info, srid)))
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/info_model.py:401]:
```suggestion
            tbl.append((QApplication.translate("DBManagerPlugin", "Spatial ref:"), "%s (%d)" % (sr_info, srid)))
```

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 7]
```diff
@@ -450,7 +463,7 @@ def spatialInfo(self):
...
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/info_model.py:466]:
```suggestion
            tbl.append((QApplication.translate("DBManagerPlugin", "Spatial ref:"), "%s (%d)" % (sr_info, srid)))
```

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 8]
```diff
@@ -247,7 +247,6 @@ def __init__(self, uri, connection):
         # self.user = uri.username() or os.environ.get('USER')
         # self.passwd = uri.password()
         self.host = uri.host()
-
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/postgis/connector.py:250]: unwanted change ?

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 9]
```diff
@@ -396,7 +395,15 @@ def fieldTypes(self):
             "date", "time", "timestamp",  # date/time
             "boolean"  # bool
         ]
-
+
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/postgis/connector.py:398]:
```suggestion

```

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 10]
```diff
@@ -396,7 +395,15 @@ def fieldTypes(self):
             "date", "time", "timestamp",  # date/time
             "boolean"  # bool
         ]
-
+        
+    def getDatabaseInfo(self):
+        """ db privileges: (can create schemas, can create temp. tables) """
```
Comment (nicogodet) [inline python/plugins/db_manager/db_plugins/postgis/connector.py:400]: copy-paste leftovers ?

**Review by nicogodet — CHANGES_REQUESTED** [review]

**Review by npltr62 — COMMENTED** [review]
@nicogodet done

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 11]
```diff
@@ -91,12 +98,18 @@ def toHtml(self):
...
         if general_info is None:
             pass
         else:
             ret.append(HtmlSection(QApplication.translate("DBManagerPlugin", 'General info'), general_info))
+        # database information
+        database_info = self.databaseInfo()
+        if database_info is None:
+            pass
+        else:
+            ret.append(HtmlSection(QApplication.translate("DBManagerPlugin", 'Database informations'), database_info))
```
Comment (Gustry) [inline python/plugins/db_manager/db_plugins/info_model.py:112]:
```suggestion
        if database_info is not None:
            ret.append(HtmlSection(QApplication.translate("DBManagerPlugin", 'Database information'), database_info))
```

"information" is always singular. Also simplifying the if condition a little bit.
We could even skip the "is not not None", up to you.

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 12]
```diff
@@ -33,7 +33,14 @@ def __init__(self, db):
 
     def __del__(self):
         self.db = None
-
+        
+    def databaseInfo(self):
+        info = self.db.connector.getDatabaseInfo()
+        tbl = [
+            (QApplication.translate("DBManagerPlugin", "Database description: "), info[0])
```
Comment (Gustry) [inline python/plugins/db_manager/db_plugins/info_model.py:40]:
```suggestion
            (QApplication.translate("DBManagerPlugin", "Database description:" + " "), info[0])
```

I think it's low probability that translators see this trailing whitespace when they translate

**Review by Gustry — APPROVED** [review]
Looks good to me.
There is some flake8 issues in the CI.

File: python/plugins/db_manager/db_plugins/info_model.py (hunk 12 above)
Comment (nicogodet) [reply python/plugins/db_manager/db_plugins/info_model.py:40]: In that case it should be 
```python
(QApplication.translate("DBManagerPlugin", "Database description:") + " ", info[0])
``` 
?

File: python/plugins/db_manager/db_plugins/postgis/connector.py [hunk 13]
```diff
@@ -396,7 +396,15 @@ def fieldTypes(self):
...
-
+        
+    def getDatabaseInfo(self):
+        """ db description: get comments of current database """
+        sql = "SELECT pg_catalog.shobj_description(d.oid, 'pg_database') AS description from pg_catalog.pg_database d where datname = current_database()"
+        c = self._execute(None, sql)
+        res = self._fetchone(c)
+        self._close_cursor(c)
+        return res
+
```
Comment (npltr62) [inline python/plugins/db_manager/db_plugins/postgis/connector.py:407]: fix indentation

File: python/plugins/db_manager/db_plugins/info_model.py [hunk 14]
```diff
@@ -33,7 +33,14 @@ def __init__(self, db):
 
     def __del__(self):
         self.db = None
-
+
```
Comment (npltr62) [inline python/plugins/db_manager/db_plugins/info_model.py:36]: fix indentation

**Comment by Gustry** [comment]
@npltr62 I already approved the PR earlier. Can anyone press the GH Action button ?

**Pull request closed without merging**